File: java/app/src/main/java/com/youknow/movie/ui/movies/MoviesContract.java
package com.youknow.movie.ui.movies;

import com.youknow.domain.model.SimpleMovie;

import java.util.List;

public interface MoviesContract {

    interface View {
        void onMoviesLoaded(List<SimpleMovie> movies);
    }

    interface Presenter {
        void getMovies();
    }
}

File: kotlin-rx/app/src/main/java/com/youknow/movie/ui/movies/MoviesContract.kt
package com.youknow.movie.ui.movies

import com.youknow.domain.model.SimpleMovie

interface MoviesContract {
    interface View {
        fun showProgressBar(visibility: Int)
        fun onMoviesLoaded(movies: List<SimpleMovie>)
        fun onError(msgResId: Int)
        fun hideError()

    }

    interface Presenter {
        fun unsubscribe()
        fun getMoviesNowPlaying()

    }
}